mosviz/viewers/mos_viewer.py

```diff
@@ -379,6 +379,74 @@ def add_data_for_testing(self, data):
         self._unpack_selection(data)
         return True
 
+    def add_data_for_jwst_testing(self, data):
```

Comment:
Sorry if this is a duplicate but I think my original comment got lost in the void somehow.

It seems like this should not really be a method of `MosVizViewer` since it's only used for testing purposes. Instead, I might suggest defining this as a function in the test module where it is called. Instead of taking `self` as the first argument, it will simply take a `MosVizViewer` instance (which technically is equivalent, but the name `self` shouldn't be used in that case).

Reply:
That seems like an easy enough start on the re-factor, i'll try it out.